iaso/api/mobile/entity.py

```diff
@@ -18,6 +21,44 @@ class LargeResultsSetPagination(PageNumberPagination):
     max_page_size = 1000
 
 
+def mobile_entity_get_queryset(request, *args, **kwargs):
+    profile = request.user.iaso_profile
+    orgunits = OrgUnit.objects.hierarchy(profile.org_units.all())
+    type_pk = request.parser_context.get("kwargs").get("type_pk", None)
+    if type_pk:
+        queryset = Entity.objects.filter(account=request.user.iaso_profile.account, entity_type__pk=type_pk)
+    else:
+        queryset = Entity.objects.filter(account=profile.account)
+    json_content = request.query_params.get("json_content", None)
+
+    # we give all entities having an instance linked to the one of the org units allowed for the current user
+    # For now, it can be filtered on OU without any access restriction
+    if orgunits:
+        queryset = queryset.filter(instances__org_unit__in=orgunits)
+    limit_date = request.query_params.get("limit_date", None)
+    if limit_date:
+        try:
+            queryset = queryset.filter(instances__updated_at__gte=limit_date)
+        except ValidationError:
+            raise Http404("Invalid Limit Date")
+    if json_content:
+        try:
+            q = jsonlogic_to_q(jsonlogic=json_content, field_prefix="json__")  # type: ignore
+            queryset = queryset.filter(q)
+        except ValidationError:
+            raise Http404("Invalid Json Content")
+
+    p = Prefetch(
+        "instances",
+        queryset=Instance.objects.filter(deleted=False).exclude(file=""),
+        to_attr="non_deleted_instances",
+    )
+
+    queryset = queryset.prefetch_related(p).prefetch_related("non_deleted_instances__form")
+
+    return queryset
```

Comment:
As we talked about, this should be rewritten to be cleaner.
Also, it introduces a bug in `/api/mobile/entitytypes/id/entities` as it now filters on the accessible org units when it should not.